### PR TITLE
287124 - API service layer

### PR DESCRIPTION
**Pull request by FujaleAhmedShaik**
### What one thing this PR does?

An API service layer will be in place to consume the external endpoint.

### Task details
- [287124 - API Service layer - FE ](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/287124)

### Other notes

- Axios has been integrated, which will give us the capability to initialize multiple clients.
- The endpoints can be consumed using the initialized client.

### How do these changes look like?

**Sample json response in the terminal**
| Sample endpoint response |
|-|

### Dev-tested in

1. Local environment  

- [Sample page](http://localhost:3000/sample2)

2. Dev environment 

- Page name/link 1. 
- ...

### CODE REVIEW

File: src/services/config/geoNetworkClient.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios, { AxiosError } from 'axios';
```
Comment (karthigayayiniram) [inline src/services/config/geoNetworkClient.ts:1]: can you move this file under src/config. in future we may have more configs coming in. lets keep all configs in a single place

File: src/services/config/geoNetworkClient.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const geoNetworkClient = axios.create({
+  baseURL: apiBaseURL,
+  headers: {
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  },
+});
+
+geoNetworkClient.interceptors.response.use(undefined, (error: AxiosError) => {
```
Comment (karthigayayiniram) [inline src/services/config/geoNetworkClient.ts:12]: can you also intercept the success and handle the message notification for accessibility?

**Review by ganesh-rajamahanthy — APPROVED** [review]

File: src/services/config/geoNetworkClient.ts (hunk 1 above)
Comment (FujaleAhmedShaik) [reply src/services/config/geoNetworkClient.ts]: The folder has been moved to the `src` folder

File: src/services/config/geoNetworkClient.ts (hunk 2 above)
Comment (FujaleAhmedShaik) [reply src/services/config/geoNetworkClient.ts]: The request and response interceptor has been updated now.

**Review by karthigayayiniram — APPROVED** [review]

**Pull request merged**